component/src/main/java/org/wso2/carbon/identity/authenticator/FoursquareAuthenticator.java

```diff
@@ -30,322 +30,367 @@
+	}
+
+	/**
+	 * Always return false as there is no ID token in Foursquare OAuth.
+	 *
+	 * @param authenticatorProperties Authenticator properties.
+	 * @return False
+	 */
+	@Override
+	protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
+		return false;
+	}
+
+	@Override
+	public String getFriendlyName() {
+		return FoursquareAuthenticatorConstants.FOURSQUARE_CONNECTOR_FRIENDLY_NAME;
+	}
+
+	@Override
+	public String getName() {
+		return FoursquareAuthenticatorConstants.FOURSQUARE_CONNECTOR_NAME;
+	}
+
+	/**
+	 * Get configuration properties.
+	 *
+	 * @return Properties list.
+	 */
+	@Override
+	public List<Property> getConfigurationProperties() {
+
+		List<Property> configProperties = new ArrayList<Property>();
+
+		Property clientId = new Property();
+		clientId.setName(OIDCAuthenticatorConstants.CLIENT_ID);
+		clientId.setDisplayName("Client Id");
+		clientId.setRequired(true);
+		clientId.setDescription("Enter Foursquare client identifier value");
+		clientId.setDisplayOrder(0);
+		configProperties.add(clientId);
+
+		Property clientSecret = new Property();
+		clientSecret.setName(OIDCAuthenticatorConstants.CLIENT_SECRET);
+		clientSecret.setDisplayName("Client Secret");
+		clientSecret.setRequired(true);
+		clientSecret.setConfidential(true);
+		clientSecret.setDescription("Enter Foursquare client secret value");
+		clientSecret.setDisplayOrder(1);
+		configProperties.add(clientSecret);
+
+		Property callbackUrl = new Property();
+		callbackUrl.setDisplayName("Callback URL");
+		callbackUrl.setName(IdentityApplicationConstants.OAuth2.CALLBACK_URL);
+		callbackUrl.setDescription("Enter the callback URL");
+		callbackUrl.setDisplayOrder(2);
+		configProperties.add(callbackUrl);
+
+		Property profileVersion = new Property();
+		profileVersion.setDisplayName("Profile Version");
+		profileVersion.setName(FoursquareAuthenticatorConstants.PROFILE_VERSION);
+		profileVersion.setDescription("Enter the profile version");
+		profileVersion.setDisplayOrder(3);
+		configProperties.add(profileVersion);
+
+		return configProperties;
+	}
+
+	/**
+	 * This method are overridden for extra claim request to Foursquare end-point.
+	 *
+	 * @param request  the http request
+	 * @param response the http response
+	 * @param context  the authentication context
+	 * @throws AuthenticationFailedException
+	 */
+	@Override
+	protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+	                                                 AuthenticationContext context)
+			throws AuthenticationFailedException {
+		try {
+			Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+			String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+			String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+			String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-foursquare/pull/2/commits/58520246ef2e797650ce96df5d6f5fbde73cfa2d